plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStoreRepositoryTests.java

```diff
@@ -188,31 +182,7 @@ public void testSnapshotWithLargeSegmentFiles() throws Exception {
             .put(BUCKET.getKey(), "bucket")
             .put(CLIENT_NAME.getKey(), "test")
             .build());
-
-        final String index = "index-no-merges";
-        createIndex(index, Settings.builder()
-            .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
-            .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)
-            .build());
-
-        final int nbDocs = 10_000;
-        try (BackgroundIndexer indexer = new BackgroundIndexer(index, "_doc", client(), nbDocs)) {
-            waitForDocs(nbDocs, indexer);
-        }
-
-        flushAndRefresh(index);
-        ForceMergeResponse forceMerge = client().admin().indices().prepareForceMerge(index).setMaxNumSegments(1).get();
-        assertThat(forceMerge.getSuccessfulShards(), equalTo(1));
-        assertHitCount(client().prepareSearch(index).setSize(0).setTrackTotalHits(true).get(), nbDocs);
-
-        assertSuccessfulSnapshot(client().admin().cluster().prepareCreateSnapshot(repository, "snapshot")
-            .setWaitForCompletion(true).setIndices(index));
-
-        assertAcked(client().admin().indices().prepareDelete(index));
-
-        assertSuccessfulRestore(client().admin().cluster().prepareRestoreSnapshot(repository, "snapshot").setWaitForCompletion(true));
-        ensureGreen(index);
-        assertHitCount(client().prepareSearch(index).setSize(0).setTrackTotalHits(true).get(), nbDocs);
+        runSnapshotWithLargeSegmentFilesTest(repository, randomLongBetween(10_000L, 20_000L));
```

Comment:
Why do we need this override, it seems this test will work fine if it's just inherited from the parent because those settings above will be supplied by the `repositorySettings` in the GCS test class exactly the same way? :)

Reply:
The test will run fine but there is a chance that a low chunk size value got randomly picked, making the test even slower that it is now.

Thinking about this again, I think we could avoid the override as you suggested and also remove the random `chunk_size` in `ESBlobStoreRepositoryIntegTestCase#repositorySettings()` for cloud based repository implementations (and move it to `FsBlobStoreRepositoryIT`) and just run with the default chunk size. Finding a random chunk size value that suits all implementations is not easy and just add more randomization to the tests that are already slow. WDYT?

Reply:
I pushed bb030f08b804489e24f64d0d904b8a30c05ee079

Reply:
Thanks I like it :)